src/renderer/mesh_storage.zig

```diff
@@ -751,6 +777,7 @@ pub fn updateMeshes(targetTime: i64) void { // MARK: updateMeshes()
 		if(getMeshAndIncreaseRefCount(pos)) |mesh| {
 			defer mesh.decreaseRefCount();
 			mesh.updateBlock(blockUpdate.x, blockUpdate.y, blockUpdate.z, blockUpdate.newBlock);
+			triggerOnUpdate(.place, blockUpdate.x, blockUpdate.y, blockUpdate.z, blockUpdate.newBlock);
```

Comment:
A block update can be both a break and a place, so you should probably also trigger the `break` action for the old block, like you already did for the server code.

Reply:
Convention here is that client side break is called on old block in `updateBlockAndSendUpdate`, while old block is still in data while here client place event here is called after new block was inserted into data.

Reply:
`updateBlockAndSendUpdate` is only called when you break or place a block locally.
The `blockUpdateList` present here is also updated on block updates from the server, so ideally you'd only call it here to avoid calling it twice.

Reply:
Wait, so `updateMeshes` is called also on server side?

Reply:
Oh, no, thats not what you ment
So updateMeshes is also called when other player destroys a block?

Reply:
yes, second one is correct.